php/Exchange.php

```diff
@@ -75,6 +75,7 @@ class Exchange {
     public $curlopt_interface = null;
     public $timeout = 10000; // in milliseconds
     public $proxy = '';
+    public $proxy_agent_url = null; // SOCKS5 & HTTP(S) supported, i.e.  socks5://127.0.0.1:8000 or http://127.0.0.1:443
```

Comment:
@ttodua we don't have  this `proxy_agent_url` name anywhere else 🤔

Reply:
Can we also support `http_proxy` ?

Reply:
I had explanation in my topcomment here, why I've named quite random different name, like `proxy_agent_url`. the reason is that we don't have an unified scheme of proxies, that's why I refrained from using some unified names, and instead, at this stage, i've made the name specifically distinct for php.

again, we need to write down and unified scheme of proxies and their names and their unified functionalities, after that we can name them accordingly across langs.
lmk your thoughts.

Reply:
i've added support for i.

Reply:
again, to read from `keys.json` (as it's done in test.js/test.py only, i've not done it in test.php yet, waiting for Igor's review of tests, hoping they might get merged).

Reply:
if we want to support `httpProxy/httpsProxy` in exchange files (instead of parsing & patching from `test.js/py/php` let me know and I'll amend all langs).

Reply:
anyway, i've added support in `test.php`s too. if there will be conflicts with other PR, i'll fix it.